src/mod_muc_db.erl

```diff
@@ -0,0 +1,46 @@
+-module(mod_muc_db).
+-include("mod_muc.hrl").
+
+%% Defines which ODBC pool to use
+%% Parent host of the MUC service
+-type server_host() :: ejabberd:server().
+
+%% Host of MUC service
+-type muc_host() :: ejabberd:server().
+
+%% User's JID. Can be on another domain accessable over FED.
+%% Only bare part (user@host) is important.
+-type client_jid() :: ejabberd:jid().
+
+-type room_opts() :: [{OptionName :: atom(), OptionValue :: term()}].
+
+
+
+%% Called when MUC service starts or restarts for each domain
+-callback init(server_host(), ModuleOpts :: list()) -> ok.
+
+-callback store_room(server_host(), muc_host(), mod_muc:room(), room_opts()) ->
+    {aborted, _} | {atomic, _}.
```

Comment:
These return values stand out, as you made other returns non-mnesia-specific (`{ok, _}`, `{error, _}`). I'd suggest doing the same with `store_room/4`.

Reply:
yeah, but mod_muc_room just ignores output value for now

Reply:
The other point is that other backends, for example a future Riak backend, will have to replicate alien Mnesia interface.